cmd/kube-proxy/app/server.go

```diff
@@ -93,20 +97,45 @@ func (s *ProxyServer) Run(_ []string) error {
 	serviceConfig := config.NewServiceConfig()
 	endpointsConfig := config.NewEndpointsConfig()
 
-	protocol := iptables.ProtocolIpv4
-	if s.BindAddress.To4() == nil {
-		protocol = iptables.ProtocolIpv6
+	protocol := utiliptables.ProtocolIpv4
+	if net.IP(s.BindAddress).To4() == nil {
+		protocol = utiliptables.ProtocolIpv6
 	}
-	loadBalancer := userspace.NewLoadBalancerRR()
-	proxier, err := userspace.NewProxier(loadBalancer, s.BindAddress, iptables.New(exec.New(), protocol), s.PortRange)
-	if err != nil {
-		glog.Fatalf("Unable to create proxer: %v", err)
+
+	var proxier proxy.ProxyProvider
+	var endpointsHandler config.EndpointsConfigHandler
+
+	// guaranteed false on error, error only necessary for debugging so we can ignore
+	shouldUseIptables, _ := iptables.ShouldUseIptablesProxier()
```

Comment:
Ah, I was. Not sure why I took that out putting that back in.

On Mon, Aug 10, 2015 at 6:26 PM, Tim Hockin <notifications@github.com>
wrote:

> In cmd/kube-proxy/app/server.go
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/9210#discussion_r36692357>
> :
>
> >  	}
> > -	loadBalancer := userspace.NewLoadBalancerRR()
> > -	proxier, err := userspace.NewProxier(loadBalancer, s.BindAddress, iptables.New(exec.New(), protocol), s.PortRange)
> > -	if err != nil {
> > -		glog.Fatalf("Unable to create proxer: %v", err)
> > +
> > +	var proxier proxy.ProxyProvider
> > +	var endpointsHandler config.EndpointsConfigHandler
> > +
> > +	// guaranteed false on error, error only necessary for debugging so we can ignore
> > +	shouldUseIptables, _ := iptables.ShouldUseIptablesProxier()
>
> shouldn't you log the error?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/9210/files#r36692357>
> .
>